src/components/search-panel/SearchPanel.tsx

```diff
@@ -42,6 +44,62 @@ export type SearchPanelState = {
 	stream: string[];
 };
 
+export type Result = EventTreeNode | EventMessage;
+
+export type SearchHistory = {
```

Comment:
Maybe we should store requests and results together ? e.g. `{ request: SearchRequest, result: SearchResult, id: .... }`